core/src/main/scala/com/salesforce/op/filters/RawFeatureFilter.scala

```diff
@@ -191,87 +191,277 @@ class RawFeatureFilter[T]
    * @param correlationInfo  info needed to determine feature to drop based on null label-leakage correlation
-   * @return a list of feature names that should be dropped and a map of map keys that should be dropped
-   *         Map(featureName -> key)
+   * @return metrics computed by raw feature filter tests for each feature
    */
-  private[op] def getFeaturesToExclude(
+  private[op] def getRawFeatureFilterMetrics(
     trainingDistribs: Seq[FeatureDistribution],
     scoringDistribs: Seq[FeatureDistribution],
     correlationInfo: Map[FeatureKey, Map[FeatureKey, Double]]
-  ): (Seq[String], Map[String, Set[String]]) = {
+  ): Seq[RawFeatureFilterMetrics] = {
+
+    val featureSize: Int = trainingDistribs.length
+
+    val trainingFillRates: Seq[Double] = trainingDistribs.map(_.fillRate())
+
+    val trainingNullLabelAbsoluteCorrs: Seq[Option[Double]] =
+      if (correlationInfo.isEmpty) Seq.fill(featureSize)(None)
+      else {
+        val absoluteCorrs =
+          for {
+            distrib <- trainingDistribs
+          } yield correlationInfo.values.map(_.get(distrib.featureKey))
+        absoluteCorrs.flatten
+      }
+
+    // combines metrics computed for each feature into RawFeatureMetrics class
+    def combineRawFeatureFilterMetrics(
+      traininingDistribs: Seq[FeatureDistribution],
+      trainingFillRates: Seq[Double],
+      trainingNullLabelAbsoluteCorrs: Seq[Option[Double]],
+      scoringFillRates: Seq[Option[Double]],
+      jsDivergences: Seq[Option[Double]],
+      fillRateDiffs: Seq[Option[Double]],
+      fillRatioDiffs: Seq[Option[Double]]
+    ): Seq[RawFeatureFilterMetrics] = {
+
+      trainingDistribs.map(_.name)
+        .zip(trainingFillRates)
+        .zip(trainingNullLabelAbsoluteCorrs)
+        .zip(scoringFillRates)
+        .zip(jsDivergences)
+        .zip(fillRateDiffs)
+        .zip(fillRatioDiffs)
+        .map {
+          case ((((((name, trainingFillRate), trainingNullLabelAbsoluteCorr),
+          scoringFillRate), jsDivergence), fillRateDiff), fillRatioDiff) =>
+            RawFeatureFilterMetrics(
+              name, trainingFillRate, trainingNullLabelAbsoluteCorr,
+              scoringFillRate, jsDivergence, fillRateDiff, fillRatioDiff)
+        }
+    }
+
+    if (scoringDistribs.isEmpty) {
+
+      val scoringFillRates = Seq.fill(featureSize)(None)
+      val jsDivergences = Seq.fill(featureSize)(None)
+      val fillRateDiffs = Seq.fill(featureSize)(None)
+      val fillRatioDiffs = Seq.fill(featureSize)(None)
+
+      val rawFeatureFilterMetrics = combineRawFeatureFilterMetrics(
+        trainingDistribs, trainingFillRates, trainingNullLabelAbsoluteCorrs,
+        scoringFillRates, jsDivergences, fillRateDiffs, fillRatioDiffs
+      )
+      rawFeatureFilterMetrics
+
+    } else {
+
+      require(scoringDistribs.length == featureSize, "scoring and training features must match")
+
+      val scoringFillRates = scoringDistribs.map(distrib => Some(distrib.fillRate()))
```

Comment:
Please use `Option` instead of `Some` (here and below). Last thing we want is to end up with `Some(null)` ;)